### PR TITLE
create delete work route and testing it

### PR DESCRIPTION
**Pull request by abdulrahman-2020**
Relates #71

- create delete work controller
- testing delete from database

### CODE REVIEW

File: server/routes/index.js [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 
 const {
-  getHomeData, getWorkPerPage, getProfileInfo, addReview, subServices,
-  // checkAuth,
+  getHomeData, getWorkPerPage, getProfileInfo, addReview, subServices, deleteWork, checkAuth,
 } = require('../controllers');
 
 // router.use(checkAuth);
-router.get('/providers/:id', getProfileInfo);
+router.get('/providers/:id', checkAuth, getProfileInfo);
```
Comment (LinaYahya) [inline server/routes/index.js:8]:
```suggestion
router.get('/providers/:id', getProfileInfo);
```

File: server/test/profile.test.js [hunk 2]
```diff
@@ -51,6 +55,7 @@ describe('Routes Tests GET /api/v1/providers/:id', () => {
   test('Testing for /api/v1/providers/1 to ensure that the response contains locations data relates to userId:1', (done) => {
     supertest(app)
       .get('/api/v1/providers/1')
+      .set('Cookie', [`token= ${process.env.token}`])
```
Comment (LinaYahya) [inline server/test/profile.test.js:58]:
```suggestion
```
Remove this line for all GET /providers/:id tests

File: server/controllers/work/deleteWork.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const { paramsValidation } = require('../../utils/validation');
+
+const deleteWork = async (req, res, next) => {
+  try {
+    const { id } = await paramsValidation.validateAsync(req.params);
+    const work = await Work.findByPk(id);
+    if (!work) {
+      throw customError('work does not exist', 400);
+    }
+    await work.destroy({ where: { id } });
```
Comment (LinaYahya) [inline server/controllers/work/deleteWork.js:12]: you need to check first if the user who send the request the same how own the work

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: server/controllers/work/deleteWork.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+const { Work } = require('../../database/models');
+const { customError } = require('../errors');
+const { paramsValidation } = require('../../utils/validation');
+
+const deleteWork = async (req, res, next) => {
+  try {
+    const { userId } = req.body;
```
Comment (LinaYahya) [inline server/controllers/work/deleteWork.js:7]: actually work has a userID as `userId`,
So we'll get userid from it not from request body

File: server/test/work.test.js [hunk 5]
```diff
@@ -28,6 +28,17 @@ describe('Routes Tests GET /api/v1/work/:providerId', () => {
       });
   });
 
+  test('Testing for delete work /api/v1/work/1 and get status 401', (done) => {
```
Comment (LinaYahya) [inline server/test/work.test.js:31]: add a test to test when user want to chack a work how doesn't own

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by RandSohail — APPROVED** [review]

**Review by devstm — APPROVED** [review]

**Review by AbdallahGot — APPROVED** [review]

**Pull request merged**